Automattic/jetpack
# Search: fix search submenu visibility

#### Changes proposed in this Pull Request:
The filter to change to visibility of Search submenu is loaded too late, and might not be executed at all, which leads to an issue that the Search submenu is shown regardless.

The PR moved the loading of compatibility file to the very beginning of the search package initialization.

Sorry the issue took few PRs to settle.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- [Create a JN site](https://jurassic.ninja/specialops/) with Jetpack Beta with branch `fix/search-menu-visibility`
![image](https://user-images.githubusercontent.com/1425433/160721496-e5a67c2f-ff15-4c35-97cb-a3b91ed6c512.png)

- Ensure Search submenu is not shown
- Connect site and purchase a Search subscription
- Ensure Search submenu is shown
